spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/GatewayMetricsFilter.java

```diff
@@ -53,6 +70,14 @@ public int getOrder() {
 		return NettyWriteResponseFilter.WRITE_RESPONSE_FILTER_ORDER + 1;
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext context) throws BeansException {
+		if (initialized.compareAndSet(false, true)) {
+			this.compositeTagsProvider = context.getBean("defaultGatewayTagsProvider",
```

Comment:
Why do this?  It should be included in the list passed into the constructor

Reply:
It is for using existing bean declared in `GatewayMetricsAutoConfiguration.java`.
If it is not needed, I'll change this as you said.

Reply:
Yeah I dont think it would be needed